scripts/add_extra_stores.py

```diff
@@ -0,0 +1,219 @@
+
+def attach_stores(n, costs):
+    elec_opts = snakemake.config['electricity']
+    carriers = elec_opts['extendable_carriers']['Store']
+
+    _add_missing_carriers_from_costs(n, costs, carriers)
+
+    buses_i = n.buses.index
+    bus_sub_dict = {k: n.buses[k].values for k in ['x', 'y', 'country']}
+
+    if 'H2' in carriers:
+        h2_buses_i = n.madd("Bus", buses_i + " H2", carrier="H2", **bus_sub_dict)
+
+        n.madd("Store", h2_buses_i,
+               bus=h2_buses_i,
+               e_nom_extendable=True,
+               e_cyclic=True,
+               capital_cost=costs.at["hydrogen storage", "capital_cost"])
+
+        n.madd("Link", h2_buses_i + " Electrolysis",
+               bus0=buses_i,
+               bus1=h2_buses_i,
+               p_nom_extendable=True,
+               efficiency=costs.at["electrolysis", "efficiency"],
+               capital_cost=costs.at["electrolysis", "capital_cost"])
+
+        n.madd("Link", h2_buses_i + " Fuel Cell",
+               bus0=h2_buses_i,
+               bus1=buses_i,
+               p_nom_extendable=True,
+               efficiency=costs.at["fuel cell", "efficiency"],
+               #NB: fixed cost is per MWel
+               capital_cost=costs.at["fuel cell", "capital_cost"] * costs.at["fuel cell", "efficiency"])
+
+    if 'battery' in carriers:
+        b_buses_i = n.madd("Bus", buses_i + " battery", carrier="battery", **bus_sub_dict)
+
+        n.madd("Store", b_buses_i,
+               bus=b_buses_i,
+               e_cyclic=True,
+               e_nom_extendable=True,
+               capital_cost=costs.at['battery storage', 'capital_cost'])
+
+        n.madd("Link", b_buses_i + " charger",
+               bus0=buses_i,
+               bus1=b_buses_i,
+               efficiency=costs.at['battery inverter', 'efficiency'],
+               capital_cost=costs.at['battery inverter', 'capital_cost'],
+               p_nom_extendable=True)
+
+        n.madd("Link", b_buses_i + " discharger",
+               bus0=b_buses_i,
+               bus1=buses_i,
+               efficiency=costs.at['battery inverter','efficiency'],
+               capital_cost=costs.at['battery inverter', 'capital_cost'],
+               p_nom_extendable=True)
+
+def add_nice_carrier_names(n):
+    nice_names = pd.Series(snakemake.config['plotting']['nice_names'])
+    n.carriers['nice_names'] = nice_names[n.carriers.index]
+
+
+if __name__ == "__main__":
+    # Detect running outside of snakemake and mock snakemake for testing
+    if 'snakemake' not in globals():
+        from vresutils.snakemake import MockSnakemake, Dict
+
+        snakemake = MockSnakemake(output=['networks/elec_s_5.nc'])
+        snakemake.input = snakemake.expand(
+            Dict(network='networks/elec_s_5.nc',
+                 tech_costs='data/costs.csv'))
+
+    logging.basicConfig(level=snakemake.config['logging_level'])
+
+    n = pypsa.Network(snakemake.input.network)
+    Nyears = n.snapshot_weightings.sum()/8760.
```

Comment:
I don't think we need this here.

Reply:
You mean the `Nyears`? It is used below for the cost assumptions

Reply:
yes, I was absolutely wrong here.